### DISCUSSION TITLE
Cannot build px4 firmware

### DISCUSSION THREAD

**Discussion by MartinSegaert** [ctu-mrs/mrs_uav_system#153] [Q&A]
Hi, 
We recently realised that we have to update our px4 firmware to the newest one (v1.13.2).  (https://github.com/ctu-mrs/mrs_uav_system/discussions/149#discussioncomment-4707267). I managed to build the firmware (make px4_fmu-v5_default) on one of my NUCs but not on the other one. I keep getting the same error:

"/home/nuc2/git/simulation/px4_firmware/srx/lib/rc/sbus.cpp:61:10 fatal error: asm-generic/termbits.h: No such file or directory"

![Image1](https://user-images.githubusercontent.com/62548016/213449388-cd1db982-210f-46ea-bad6-9c8bfbed9a09.png)

The NUC on which the build works has CMake version 3.20.2. The NUC on which the build does not work has CMake version 3.25.0. Could the version of CMake be the reason why it fails? If that's the case, is CMake version 3.20.2 alright for the good working of the CTU packages?

Since I have successfully built the firmware on one NUC, I could take that file to update the pixhawk that goes with the NUC where it doesn't work. Would that work? Or would it give problems in simulation on the NUC were the build doesn't work ?

Thanks

**Comment by dagar** [comment]
Which version of GCC are you using? If you build again from a clean state it should earlier in the cmake configure output.

**Reply by MartinSegaert** [reply]
Thanks for your response.

My GCC version is 9.4.0:

nuc2@uav2:~$ gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

What do you mean by building again from a clean state ?

**Reply by dagar** [reply]
> What do you mean by building again from a clean state ?

Like a fresh checkout, or `make distclean` (or even just `rm -rf build/`). Then the full cmake configure will run again on the next build.

**Reply by dagar** [reply]
I think I see the problem, `-D__PX4_LINUX` is set in the compile line, but aren't you trying to build px4_fmu-v5_default?

**Reply by MartinSegaert** [reply]
Yes, I am trying to build px4_fmu-v5_default. I typed "make px4_fmu-v5_default". 

I deleted the build folder every time I tried again. I also tried "make distclean" once as mentioned in this link (in "General build errors") https://docs.px4.io/main/en/dev_setup/building_px4.html. But it did not work better.

**Reply by MartinSegaert** [reply]
Why is it a problem that `-D__px4_linnux` is set in the compile line? And how can I resolve this?

Thanks

**Reply by gugafelds** [reply]
Hi @MartinSegaert, have you already solved that? I'm having the same problem

**Reply by MartinSegaert** [reply]
Hi,
No, I did not solve the problem. I actually stopped working on this project and I am restarting now.

**Reply by MartinSegaert** [reply]
@gustavotbarros
I resolved the problem by doing a fresh ubuntu installation on my NUC. Without upgrading cmake to the newest version I was able to build the pixhawk firmware.

**Reply by gugafelds** [reply]
@MartinSegaert Hi, thanks for the response. Which version from cmake are you using now?

**Reply by MartinSegaert** [reply]
cmake version  3.16.3
It is the one that got installed when running `sudo apt-get install cmake`
